### ISSUE TITLE
Return status 500 instead of abort

### ISSUE THREAD

**Issue by maskit** [apache/trafficserver#10082]
`ink_release_assert` is used at over 400 places. Although having core files is helpful for debugging, it kills all connections abruptly  and it's not nice to clients. It'd be nice if we could just return status 500 and save connections/requests that don't face any issues.

**Comment by jpeach** [comment]
`ink_release_assert` should be used for internal invariants (i.e. when you do something that never fails and you want to document that it never fails). If `ink_release_assert` ever fails, it ought to be a serious bug.

**Comment by maskit** [comment]
Yes I'd use it like that, but some of `ink_release_assert` are casually used like "I want some hint (coredump) if this check fails".

**Comment by bryancall** [comment]
I am closing this as a stale enhancement request with no clear path forward. The suggestion is to make process-aborting `ink_release_assert` checks instead fail the affected transaction with a 500 so unrelated connections are not killed. As noted in the discussion, `ink_release_assert` is intended for internal invariants that should never fail and that indicate a serious bug when they do, so converting them wholesale is not straightforward. The behavior is unchanged on current master (`include/tscore/ink_assert.h` still routes `ink_release_assert` through `_ink_assert`, which never returns, and there are over 400 call sites), and no implementation or design proposal has materialized since 2023. If a concrete, scoped plan emerges for converting specific transaction-level invariants into recoverable error responses, please open a new issue with those specifics and we can pursue it there.
